### PR TITLE
Convert hermes to be a remote dependency

### PR DESCRIPTION
**Pull request by oguzkocer**
Depending on Hermes as a local dependency caused quite a bit of pain so far. While working on my HACK week project, which involved trying S3 as a possible Bintray replacement, I've decided to give uploading Hermes to S3 a try. I've used the below command from [maven install plugin](https://maven.apache.org/plugins/maven-install-plugin/) to generate the POM file and did some manual cleanup and uploaded to the S3 bucket. Then a simple repository addition made it possible to read the Hermes binary from S3. I've then expanded on this work and came up with https://github.com/oguzkocer/gutenberg/pull/3 which in summary adds the remote hermes binary as a runtime dependency to `react-native-bridge`, so we don't need to know anything about Hermes in WPAndroid.

```
mvn install:install-file -Dfile=./hermes-release.aar -DgroupId=org.wordpress-mobile -DartifactId=hermes-release -Dversion=0.61.5 -Dpackaging=aar -DgeneratePom=true
```

* I've opted to use `org.wordpress-mobile` as the `groupId` to make it clear that this is not an official repository. 
* I've used `0.61.5` as the `version` because Hermes versions are tied to `react-native` versions. [#](https://github.com/facebook/hermes/releases) It might be good to figure out which Hermes binary version we are using, so we can use that version number. (in some ways I also like using `react-native` version for it to make that coupling more obvious)

### CODE REVIEW

File: libs/editor/WordPressEditor/build.gradle [hunk 1]
```diff
@@ -83,15 +84,6 @@ dependencies {
...
     lintChecks 'org.wordpress:lint:1.0.1'
-
-    // We need to add the hermes library manually for binary dependency.
-    // There doesn't seem to be a good way to integrate an aar into a library dependency and Google
-    // doesn't seem to believe its worth implementing one: https://stackoverflow.com/a/20715155
-    if (gradle.ext.includedBuildGutenbergMobilePath == null) {
-        def hermesPath = './libs/hermes/'
-        debugRuntimeOnly files(hermesPath + 'hermes-debug.aar')
-        releaseRuntimeOnly files(hermesPath + 'hermes-release.aar')
-    }
```
Comment (mokagio) [inline libs/editor/WordPressEditor/build.gradle:94]: This also solve the awkward breadcrumb link discussion we had!

The best way to solve a problem is to put yourself in the position where you don't have it :clap

**Review by mokagio — COMMENTED** [review]
Tested on the Emulator ensuring I was using the binary build and not the path in `local-builds.gradle`.

The editor worked 👏 I posted [this](https://amokafullofstuff.wordpress.com/2021/03/11/test-post-with-hermes-from-s3/) with it.

Something that probably be unrelated but worth tracking. I tried to publish a new page, just to do something different. When I hit publish the app crashed with:

```
java.lang.RuntimeException:
  Debug build crash:
    Post had changes that needed to be saved when exiting the editor, and there was NOT an
    autosave pending. This means that the editor's autosave mechanism failed. This only 
    crashes on debug builds.
```

---

This is ✅ for me, but only commenting since it's still a draft.

**Comment by mchowning** [comment]
> Something that probably be unrelated but worth tracking. I tried to publish a new page, just to do something different. When I hit publish the app crashed with:
>
> java.lang.RuntimeException:
>   Debug build crash:
>     Post had changes that needed to be saved when exiting the editor, and there was NOT an
>     autosave pending. This means that the editor's autosave mechanism failed. This only 
>     crashes on debug builds.

I agree that this is unrelated to these changes. 👍  I would be curious if you could share the content of the page you were trying to publish though @mokagio .

cc: @cameronvoell

**Review by loremattei — APPROVED** [review]
Code changes make sense. 
I tried to publish a post and a page and it worked as expected. My tests were on very simple posts and pages though.

**Comment by oguzkocer** [comment]
Thank you all for taking a look at this!

@mokagio I've seen that crash before. It's definitely not related to anything in the composite build project and I don't think it's even related to Gutenberg. I think there is a case that's missed in the upload code and that's why that DEBUG crash was added. If it keeps happening, feel free to report a separate issue and maybe ask if it can be handled as part of Groundskeeping.

**Comment by mokagio** [comment]
@mchowning I got it, and was able to reproduce today, trying to publish this page

Screen Shot 2021-03-12 at 3 30 45 pm

The activity crashed, but the page was saved as a draft which is handy. In my second reproduction attempt, the app actually showed a toast about it with a publish button. A nice way to recover.

Screen Shot 2021-03-12 at 3 33 33 pm

**Pull request merged**